### ISSUE TITLE
.find() response should be more clear if nothing is found

### ISSUE THREAD

**Issue by facultymatt** [FacultyCreative/ngActiveResource#2]
Currently `.find()` will return a blank model if nothing is found. It would be more ideal if this method returned false, null, an empty array, or something similar.

Currently I've worked around this by using: 

```
// find sensor and redirect if not found
Sensor.find({
    id: sensorId
}).then(function(response) {
    if(!response.id) $redirect('settings_sensors_path');
    $scope.sensor = response;
});
```

But obviously checking for `.id` is not ideal.

**Comment by facultymatt** [comment]
Also strange is that `find()` will always return the first item if there is no match AND there are cached items. Would make more sense to return null as well.

**Comment by brettshollenberger** [comment]
This is a larger issue I want to work on for implementing .success and .error promise resolutions through `$q` for all methods. Thanks for bringing this up; I'll let you know when it's resolved.

**Comment by facultymatt** [comment]
Makes sense. At a basic level no results could be considered an error. However this might conflict with actual errors (ie 500, 404) from api? Needs more thought but in general this approach sounds like a good direction. 

Marking this as enhancement with no milestone.
